src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -543,6 +543,96 @@ public void UserDefinedFunctionTest(string udfExpression, string expression, boo
             }
         }
 
+        [Theory]
+        [InlineData(
+            "Point = Type({x : Number, y : Number}); distance(a: Point, b: Point): Number = Sqrt(Power(b.x-a.x, 2) + Power(b.y-a.y, 2));",
+            "distance({x: 0, y: 0}, {x: 0, y: 5})",
+            true,
+            5.0)]
+
+        // Table types are accepted
+        [InlineData(
+            "People = Type([{Id:Number, Age: Number}]); countMinors(p: People): Number = CountRows(Filter(p, Age < 18));",
+            "countMinors([{Id: 1, Age: 17}, {Id: 2, Age: 21}])",
+            true,
+            1.0)]
+        [InlineData(
+            "Numbers = Type([Number]); countEven(nums: Numbers): Number = CountRows(Filter(nums, Mod(Value, 2) = 0));",
+            "countEven([1,2,3,4,5,6,7,8,9,10])",
+            true,
+            5.0)]
+
+        // Type Aliases are allowed
+        [InlineData(
+            "CarYear = Type(Number); Car = Type({Model: Text, ModelYear: CarYear}); createCar(model:Number, year: Number): Car = {Model:model, ModelYear: year};",
+            "createCar(\"Model Y\", 2024).ModelYear",
+            true,
+            2024.0)]
+
+        // Type definitions order shouldn't matter
+        [InlineData(
+            "Person = Type({Id: IdType, Age: Number}); IdType = Type(Number); createUser(id:Number, a: Number): Person = {Id:id, Age: a};",
+            "createUser(1, 42).Age",
+            true,
+            42.0)]
+
+        // Functions accept record with more/less fields
+        [InlineData(
+            "People = Type([{Name: Text, Age: Number}]); countMinors(p: People): Number = CountRows(Filter(p, Age < 18));",
+            "countMinors([{Name: \"Bob\", Age: 21, Title: \"Engineer\"}, {Name: \"Alice\", Age: 25, Title: \"Manager\"}])",
+            true,
+            0.0)]
+        [InlineData(
+            "Employee = Type({Name: Text, Age: Number, Title: Text}); getAge(e: Employee): Number = e.Age;",
+            "getAge({Name: \"Bob\", Age: 21, Title: \"Engineer\"})",

```

Comment:
The comment says that those are tests for functions accepting records with more/less fields, but the record has the same number of fields as the type